arangod/Sharding/ShardingInfo.cpp

```diff
@@ -424,6 +424,14 @@ std::vector<std::string> const& ShardingInfo::shardKeys() const {
 
 std::shared_ptr<ShardMap> ShardingInfo::shardIds() const { return _shardIds; }
 
+std::shared_ptr<std::vector<ShardID>> ShardingInfo::shardListAsShardID() const {
+  std::vector<ShardID> testVector;
```

Comment:
why don't you start with a shared_ptr<std::vector> straight away?

Reply:
Also can we make this sorted?
The API to get a sorted shortlist is missing in other places anyways.

Reply:
@mchacki applied requested changes. Please re-review that part.